src/sqlfluff/rules/L052.py

```diff
@@ -152,10 +152,11 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
                             break
                         except ValueError:  # pragma: no cover
                             pass
-            else:
-                self.logger.error(
+            elif memory != dict(ended_statements=deque()):
```

Comment:
I think this will work the same, and it's tidier because:
* It's checking the specific item inside the collection, not the whole collection.
* It avoids constructing two temporary objects (`deque` and `dict`).
```suggestion
            elif memory["ended_statements"]:
```